test/behaviour/reasoner/verification/SoundnessVerifier.java

```diff
@@ -101,14 +114,46 @@ private boolean canExplanationBeVerified(Explanation explanation) {
     private void verifyAnswerAndCollectExplanations(ConceptMap answer, Transaction tx) {
         verifyExplainableVars(answer);
         answer.explainables().iterator().forEachRemaining(explainable -> {
-            tx.query().explain(explainable.id()).forEachRemaining(explanation -> {
+            List<Explanation> explanations = tx.query().explain(explainable.id()).toList();
+            iterate(explanations).forEachRemaining(explanation -> {
                 // This check is valid given that there is no mechanism for recursion termination given by the UX of
                 // explanations, so we do it ourselves
                 if (collectedExplanations.contains(explanation)) return;
                 else collectedExplanations.add(explanation);
                 verifyAnswerAndCollectExplanations(explanation.conditionAnswer(), tx);
                 verifyVariableMapping(answer, explainable, explanation);
             });
+
+            Pair<Conjunction, ConceptMap> deferredBindableConjunction = new Pair<>(explainable.conjunction(), answer.filter(explainable.conjunction().retrieves()));
+            assert !deferredExplanationCount.containsKey(deferredBindableConjunction) ||
+                    deferredExplanationCount.get(deferredBindableConjunction).equals(explanations.size());
+            deferredExplanationCount.put(deferredBindableConjunction, explanations.size());
```

Comment:
We can't call verifyNumberOfExplanations here because the inferredConcepts are not guaranteed to have been mapped. We instead store the conjunction and the pre-image of the mapped concepts, along with the expected explanation count.